File: backend/src/main/java/com/youtube/clone/youtube_clone/dto/UserDto.java
package com.youtube.clone.youtube_clone.dto;

import java.util.Set;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserDto {

    private String id;
    private String firstName;
    private String lastName;
    private String fullName;
    private String emailAddress;
    private String sub;
    private String channelName;
    private boolean firstTimeUser;
    private String address;
    private String phoneNumber;
    private Set<String> subscribedToUsers;
    private Set<String> subscribers;
    private Set<String> videoHistory;
    private Set<String> likedVideos ;
    private Set<String> dislikedVideos;



    public boolean getFirstTimeUser(){
        return firstTimeUser;
    }
    
}

File: frontend/youtube-clone-ui/src/app/user-dto.ts
import { UrlWithStringQuery } from "url";

export interface UserDto{
    id: string;
    sub: string;
    firstName: string;
    lastName: string;
    nickname: string;
    fullName: string;
    picture: string;
    email: string;
    channelName: string;
    address: string;
    phoneNumber: string;
    firstTimeUser: boolean;
}